controllers/articlesController.js

```diff
@@ -93,4 +82,154 @@ export default class PostController {
       });
     }
   }
+
+  /**
+   * @description It helps the user to fetch all of the created articles.
+   * @param  {object} req - The request object
+   * @param  {object} res - The response object
+   * @returns {object} It returns the request's response object
+   */
+
+  static async findAll(req, res) {
+    const { user: { id = null } = {} } = req;
+    const { page = 1, limit = 20 } = req.query;
+    try {
+      const all = await Article.findAll({
+        offset: (Number(page) - 1) * Number(limit),
+        limit,
+        order: [["createdAt", "DESC"]],
+        where: { deletedAt: null },
+        attributes: { exclude: ["deletedAt"] },
+        include: [
+          {
+            model: User,
+            as: "author",
+            attributes: ["username", "bio", "image"],
+            include: [
+              { model: User, as: "followers", attributes: ["id", "username"] }
+            ]
+          },
+          { model: User, as: "likes", attributes: ["username"], through: {} },
+          { model: Tag, attributes: ["name"], as: "tagsList", through: {} }
+        ]
+      });
+      if (all.length) {
+        const articles = all.map(each => {
+          const {
+            author: { followers, username, bio, image }
+          } = each.get();
+          const { tagsList, likes, ...article } = each.get();
+          const following = !!_.dropWhile(followers, f => f.id !== id).length;
+          return {
+            ...article,
+            likes: likes.map(like => like.username),
+            author: { following, username, bio, image },
+            favorited: !!likes.length,
+            favoritesCount: likes.length,
+            tagsList: tagsList.map(t => t.get().name || null)
+          };
+        });
+        const articlesCount = articles.length;
+        return res
+          .status(OK)
+          .json({ message: "Successful", articles, articlesCount });
```

Comment:
@Musigwa You also need to return the total number of pages and the current page such as:
```
{
 message: String, // Success message
 articles: Array, // List of articles
 articlesCount: Number, // Total number of articles
 page: Number, // Current page
 pages: Number // Total number of pages
}
```

Reply:
@oesukam, following up to what you said about the scenario where a user will try to access a non-existing page, we already have that particular error handler from the server index file. Check it out to see